management/backup.py

```diff
@@ -473,10 +473,91 @@ def list_target_files(config):
-
+	elif target.scheme == "b2":
+		api = b2_perform_authentication(target.username, target.password)
+		if api == None:
+			raise ValueError("B2 authentication failed for unknown reason.")
+		(auth_token, api_url) = api
+
+		bucket_id = b2_get_bucket_id(target.username, api_url, auth_token, target.hostname)
+		return b2_list_files(target.username, api_url, auth_token, bucket_id, target.path)
 	else:
 		raise ValueError(config["target"])
 
+def b2_perform_authentication(acc_id, app_key):
+	# Perform Authentication with B2 Storage. Returns a tuple
+	# containing the auth token and the API endpoint to use.
+	from urllib.request import Request, urlopen
+	from urllib.error import HTTPError
+	import base64
+	import json
+
+	# Formulate the authtoken according to B2 Storage API Spec
+	auth_token = "Basic" + base64.b64encode((acc_id + ":" + app_key).encode('utf-8')).decode('utf-8')
+	headers = {'Authorization': auth_token}
+
+	req = Request("https://api.backblazeb2.com/b2api/v1/b2_authorize_account", headers=headers)
+	try:
+		with urlopen(req) as res:
+			data = json.loads(res.read().decode('utf-8'))
+			return (data['authorizationToken'], data['apiUrl'])
+	except HTTPError as e:
+		e_data = json.loads(e.read().decode('utf-8'))
+		print(e_data)
+		if e_data['code'] == 'unauthorized':
+			raise ValueError("Invalid key pair, or B2 has not been enabled, or the user is suspended.")
+		elif e_data['code'] == 'missing_phone_number':
+			raise ValueError("The B2 account must have a phone number associated.")
+
+def b2_perform_request(acc_id, api_url, auth_token, method, data):
+	# Perform a request against the B2 Storage API. b2_perform_authentication must
+	# have been called first to get the appropriate API endpoint and auth token.
+	# Refer to the B2 Storage API for more details. Since the method varies, the CALLER
+	# must catch any errors.
+	from urllib.request import Request, urlopen
+	from urllib.error import HTTPError
+	import base64
+	import json
+
+	if data != None:
+		req_data = json.dumps(data).encode('utf-8')
+	else:
+		req_data = None
+
+	req = Request("{0}/b2api/v1/{1}".format(api_url, method), data=req_data, headers = { 'Authorization': auth_token})
+	with urlopen(req) as res:
+		data = json.loads(res.read().decode('utf-8'))
+		return data
+
+def b2_get_bucket_id(acc_id, api_url, auth_token, bucket_name):
+	from urllib.error import HTTPError
+	try:
+		bucket_list = b2_perform_request(acc_id, api_url, auth_token, 'b2_list_buckets', {'accountId': acc_id})['buckets']
+
+		# Will always have a len of 0 or 1 as bucket names are unique.
+		bucket_id = [x['bucketId'] for x in bucket_list if x['bucketName'] == bucket_name]
+		return bucket_id[0] if len(bucket_id) else None
+	except HTTPError as e:
+		e_data = json.loads(e.read().decode('utf-8'))
+		if e_data['code'] == 'bad_request':
+			raise ValueError("B2 Storage: Account ID does not exist")
+
+def b2_list_files(acc_id, api_url, auth_token, bucket_id, path):
```

Comment:
It seems like `b2_list_file_names` only returns 100 at a time. So to get all of them you need to do multiple requests. Also, it defaults to 100, but you can set `maxFileCount` to 1000.

This could be added as a doc comment.
```
"""
:param bucket_id: get with b2_get_bucket_id function.
:param path: where to look in the bucket for files.
API Documentation: https://www.backblaze.com/b2/docs/b2_list_file_names.html
"""
```